common/lib/xmodule/xmodule/modulestore/xml.py

```diff
@@ -629,9 +629,7 @@ def get_id(org, course, url_name):
         if not url_name:
             raise ValueError("Can't load a course without a 'url_name' "
                              "(or 'name') set.  Set url_name.")
-        # Have to use older key format here because it makes sure the same format is
-        # always used, preventing duplicate keys.
-        return CourseKey.from_string('/'.join([org, course, url_name]))
+        return CourseLocator(org=org, course=course, run=url_name)
```

Comment:
@ormsbee i am making change in this top level where course id is extracted from raw course

cc @pdpinch @annag

Reply:
@ormsbee @nedbat this change is casing many tests fails in cms and xmodule. Is this change OK